### DISCUSSION TITLE
Turbolinks support

### DISCUSSION THREAD

**Discussion by giddyos** [wire-elements/modal#143] [Q&A]
This package doesn't work with turbolinks enabled because I keep getting `Uncaught TypeError: Livewire.on is not a function`. I tried out a similar package https://github.com/lao9s/livewire-modal-twitter, and it works fine with turbolinks. Any fix?

**Comment by putera** [comment]
I had the same issue.

**Comment by Nurify** [comment]
php artisan vendor:publish --tag=livewire-ui-modal-config
'include_js' => false

in any js file:
```
show = false; 
require('vendor/wire-elements/modal/resources/js/modal');
```

**Reply by MACscr** [reply]
shouldnt this be managed by wepback? i dont have any raw js files

**Comment by dhruva81** [comment]
I am also facing same issue. Kindly help.

**Comment by nickturrietta** [comment]
Ran into this today.

I solved by setting `include_js` to `false` in the config, and using my own modified **modal.js** in which I wrapped the contents of the `init()` method with `setTimeout`. ie..


```js
init() {
    setTimeout( () => {
        this.modalWidth = this.getActiveComponentModalAttribute('maxWidthClass');

        Livewire.on('closeModal', (force = false, skipPreviousModals = 0, destroySkipped = false) => {
            this.closeModal(force, skipPreviousModals, destroySkipped);
        });

        Livewire.on('activeModalComponentChanged', (id) => {
            this.setActiveModalComponent(id);
        });
   }, 0);
}
```
